### PR TITLE
fixing versions RFC

### PR DESCRIPTION
**Pull request by tzk**
Hi there,

I cleaned up the requirements.txt a bit and synchronized with setup.py. This won't install yet on travis native and probably never will without manual package installation for their oldish system. I think I'll have a working docker/pip version to use for travis soon, currently working on that one.

Looking for comments on the versions now. These are now sorted alphabetically to facilitate comparison between setup and requirements.

Had to remove rc0 from numpy and b1 from scipy, these are not available on pypi (anymore?).

Switched scikit-learn to 0.17 for both setup.py and requirements.txt.

I removed the following packages because they are not available as well. What are these? Do we need them?

d3networkx==0.1
d3py==0.2.3

Cheers,
Daniel

### CODE REVIEW

**Comment by bgruening** [comment]
@tzk this is a nice read about this topic: https://caremad.io/2013/07/setup-vs-requirement/

d3 is probably not needed, only for notebooks and not for the core library.

**Comment by tzk** [comment]
Here we go, travis now builds a docker image and runs all tests using two containers. All installation is done via apt-get and pip, no more conda. Build time is up to 20 minutes, though.

I disabled all blacklisting for nose (previously excluded -I selector.py -I embedding.py -I obabel.py because of missing dependencies). This means the new docker installation officially includes every python dependency :)

Ready for merging when you are.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+# weblogo installation from requirements.txt complains about missing numpy, so install beforehand
+RUN pip install "numpy==1.8.0" >> install.log
+
+# scikit-learn installation from requirements.txt complains about missing scipy, so install beforehand
+RUN pip install "scipy==0.14.0" >> install.log
+
+# install from local copy of requirements.txt, changes in this file invalidate the cache
+ADD requirements.txt .
+ADD setup.py .
```
Comment (bgruening) [inline Dockerfile]: why is this needed?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+RUN pip install -r requirements.txt >> install.log
+
+RUN apt-get remove -y --purge libzmq-dev software-properties-common libc-dev libreadline-dev && \
```
Comment (bgruening) [inline Dockerfile]: This needs to move up and needs to be included in the long RUN apt-get command otherwise it does not have any effect.

**Comment by bgruening** [comment]
One general question, sorry if I missed some discussion here. Why do we need to build the container at every commit. Why not simply pull a container and start it with the recent EDeN version mounted?

Let me know if this is interesting and I can implement this.

File: Dockerfile (hunk 1 above)
Comment (tzk) [inline Dockerfile]: required because otherwise pip install requirements will fail. the update to requests is needed to install more than 25 requirements. both numpy and scipy have to be installed beforehand, otherwise scikit-learn and weblogo will fail during compilation

Comment (bgruening) [inline Dockerfile]: Oh this was referring to the setup.py file.

Comment (tzk) [inline Dockerfile]: you're right, not needed anymore

On Wed, Dec 16, 2015 at 4:56 PM, Björn Grüning <notifications@github.com>
wrote:

> In Dockerfile
> <https://github.com/fabriziocosta/EDeN/pull/59#discussion_r47793262>:
>
> > +    apt-get -qq update && \
> > +    apt-get install --no-install-recommends -y rnashapes vienna-rna
> > +
> > +# requests: system level package has a bug that makes installing from requirements.txt fail
> > +RUN pip install distribute --upgrade && \
> > +    pip install "requests==2.7.0" >> install.log
> > +
> > +# weblogo installation from requirements.txt complains about missing numpy, so install beforehand
> > +RUN pip install "numpy==1.8.0" >> install.log
> > +
> > +# scikit-learn installation from requirements.txt complains about missing scipy, so install beforehand
> > +RUN pip install "scipy==0.14.0" >> install.log
> > +
> > +# install from local copy of requirements.txt, changes in this file invalidate the cache
> > +ADD requirements.txt .
> > +ADD setup.py .
>
> Oh this was referring to the setup.py file.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fabriziocosta/EDeN/pull/59/files#r47793262>.
>



-- 
--------------------------------------------------
Daniel Maticzka, Dipl. Inf.
Lehrstuhl für Bioinformatik
Institut für Informatik
Albert-Ludwigs-Universität Freiburg

Tel: (+49) 761 203 97529
Fax: (+49) 761 203 7462
Mail: maticzkd@informatik.uni-freiburg.de
--------------------------------------------------

**Comment by tzk** [comment]
No, you're just thinking ahead. The only point of this was to have a fixed set of versions. The current purpose of using docker was simply to have a way to run the tests with the specific package versions. This wouldn't work with the oldish travis ubuntu.

Let's talk tomorrow how to go from there. Another thing we could do is to automatically push the images to docker hub once they pass the tests.
